sql/api/src/main/scala/org/apache/spark/sql/catalyst/parser/DataTypeAstBuilder.scala

```diff
@@ -218,6 +218,6 @@ class DataTypeAstBuilder extends SqlBaseParserBaseVisitor[AnyRef] {
    * Returns a collation name.
    */
   override def visitCollateClause(ctx: CollateClauseContext): String = withOrigin(ctx) {
-    string(visitStringLit(ctx.stringLit))
+    ctx.multipartIdentifier().getText
```

Comment:
This is a bit confusing. How do we turn a multi part identifier into a single string? using dot to connect? where is the implementation?

Reply:
Yeah, let's keep this as an identifier for now. In future, collation name may become multipart identifier, e.g. once we introduce support for user defined collations that are part of a catalog.

But for now, collation name is just an identifier.